main_window/competition/mapping/mapping_thread.py

```diff
@@ -116,25 +129,39 @@ def plotMap(self, latitude: float, longitude: float, radius: float, zoom: float)
         :return:
         :rtype:
         """
-        if longitude is None or latitude is None:
-            return False
+        if (None in latitudes) or (None in longitudes):
+             return False
 
-        p0 = mapbox_utils.MapPoint(latitude, longitude)  # Point of interest
+        #Calculate average of device points
+        avg_latitude = sum(latitudes)/len(latitudes)
+        avg_longitude = sum(longitudes) / len(longitudes)
+
+        #Maximum distance between points and average
+        max_lat_diff = max(map(lambda lat: abs(lat - avg_latitude), latitudes))
+        max_long_diff = max(map(lambda long: abs(long - avg_longitude), longitudes))
```

Comment:
I know we've used this syntax in the past. Check out [list comprehension](https://www.w3schools.com/python/python_lists_comprehension.asp) in case you like that syntax more.